utils/orion_funcs.py

```diff
@@ -23,25 +24,72 @@ def run_hunter_analyze(merged_df,test):
     """
     merged_df["timestamp"] = pd.to_datetime(merged_df["timestamp"])
     merged_df["timestamp"] = merged_df["timestamp"].astype(int) // 10**9
-    metrics = {column: Metric(1, 1.0)
-               for column in merged_df.columns
-               if column not in ["uuid","timestamp"]}
-    data = {column: merged_df[column]
-            for column in merged_df.columns
-            if column not in ["uuid","timestamp"]}
-    attributes={column: merged_df[column] for column in merged_df.columns if column in ["uuid"]}
-    series=Series(
+    metrics = {
+        column: Metric(1, 1.0)
+        for column in merged_df.columns
+        if column not in ["uuid", "timestamp"]
+    }
+    data = {
+        column: merged_df[column]
+        for column in merged_df.columns
+        if column not in ["uuid", "timestamp"]
+    }
+    attributes = {
+        column: merged_df[column] for column in merged_df.columns if column in ["uuid"]
+    }
+    series = Series(
         test_name=test["name"],
         branch=None,
         time=list(merged_df["timestamp"]),
         metrics=metrics,
         data=data,
-        attributes=attributes
+        attributes=attributes,
     )
-    change_points=series.analyze().change_points_by_time
-    report=Report(series,change_points)
-    output = report.produce_report(test_name="test",report_type=ReportType.LOG)
-    print(output)
+    change_points = series.analyze().change_points_by_time
+    report = Report(series, change_points)
+    if output == "text":
+        output_table = report.produce_report(
+            test_name="test", report_type=ReportType.LOG
+        )
+        print(output_table)
+    elif output == "json":
+        change_points_by_metric = series.analyze().change_points
```

Comment:
Thank you for the comment @vishnuchalla. I believe the main reason we find to have json is reusability of data and logic in other tools. If we have change points it might just be constrained to changepoint detection, if we have the whole data, the same can be used in cpt dashboards such that we can view the changepoints in graphs too. I think this is the major reason we went ahead with implementing a custom logic. Also custom logic allows us to add the buildUrl and else if we needed in the json document. Thoughts?

Reply:
Understood. So can we have both regulated with a toogle between just changepoints and the entire list of docs with changepoints? Because getting the entire data might be very huge all the times. Also can we have a follow-up story to only detect regressions within a given time range? To have something like [this](https://github.com/datastax-labs/hunter/blob/master/README.md#validating-performance-of-the-main-branch).

Reply:
I'll have this toggle and time range implemented along with the next PR, as already the current PR #26  has too many tasks going on.